ghzdac/correction.py

```diff
@@ -1165,10 +1188,17 @@ def DACifyFT(self, signal, t0=0, n=8192, offset=0, nfft=None, loop=False,
                 #decay times:
                 # add to qubit registry the following keys:
                 # settlingAmplitudes=[-0.05]  #relative amplitude
-                # settlingRates = [0.012]    #rate is in GHz, (1/ns)
+                # settlingRates = [0.01]    #rate is in GHz, (1/ns)
                 if np.alen(decayRates):
-                    freqs = 2j*np.pi*freqs
-                    precalc /= (1.0 + np.sum(decayAmplitudes[:, None] * freqs[None, :] / (freqs[None, :] + decayRates[:, None]), axis=0))
+                    precalc /= (1.0 + np.sum(decayAmplitudes[:, None] * 2j*np.pi*freqs[None, :] / (2j*np.pi*freqs[None, :] + decayRates[:, None]), axis=0))
+
+                #reflections:
+                # add to qubit registry the following keys:
+                # reflectionAmplitudes=[0.05]  #relative amplitude
+                # reflectionRates = [0.01]    #rate is in GHz, (1/ns)
+                if np.alen(reflectionRates):
+                    for rate,amplitude in zip(reflectionRates,reflectionAmplitudes):
+                        precalc /= (1.0 - amplitude) / (1.0-amplitude*np.exp(-2j*np.pi*freqs/rate))
```

Comment:
This is a good one. For deconvolution, you divide by the impulse response function.
Suppose X is an ideal pulse, H the impulse response of a piece of cable (with reflection, settling etc). To get X at the end of the cable you need to send Y = X/H. 

While dividing is the fully correct approach, you can approximate the settling case by adding in a particular way. This is what Max did. It should be fine as long as the amplitudes are around a few percent. I can fix it, but it won't matter in practice.